test/integration/httproute_webhook_test.go

```diff
@@ -210,16 +210,20 @@ func TestHTTPRouteValidationWebhookExpressionsRouter(t *testing.T) {
 }
 
 // setUpEnvForTestingHTTPRouteValidationWebhook sets up the environment for testing HTTPRoute validation webhook,
-// it sets it globally. See https://github.com/Kong/kubernetes-ingress-controller/issues/4621 for more details.
+// it sets it only for object applied to namespace specified as argument.
 func setUpEnvForTestingHTTPRouteValidationWebhook(ctx context.Context, t *testing.T) (
 	namespace string,
 	gatewayClient *gatewayclient.Clientset,
 	managedGateway *gatewayv1beta1.Gateway,
 	unmanagedGateway *gatewayv1beta1.Gateway,
 ) {
+	ns, cleaner := helpers.Setup(ctx, t, env)
+	namespace = ns.Name
```

Comment:
I'm not sure I follow. We accept a `namespace` parameter yet we don't use it and we set it to the name of the namespace that's created by `helpers.Setup` 🤔

Reply:
This function returns it. Named parameter served as docs in this case

Reply:
Oh 🤦